### PR TITLE
try catch in parallel worker

### PR DESCRIPTION
**Pull request by abuts**
Let's see if it makes any difference or tell us something new

### CODE REVIEW

File: admin/worker_4tests.m.template [hunk 1]
```diff
@@ -13,14 +13,62 @@ function [ok,err_mess,je]=worker_4tests(worker_controls_string)
...
+        pid = int64(feature('getpid'));
+        log_file = sprintf('WORKER_V2_pid%i_PackageInit_failure.log',pid);
+        write_fail_log(log_file,err_mess)
+        return;
+    end
+end
+DO_LOGGING = false;
+DO_DEBUGGING = false;
+try
+    [ok, err_mess,je] = parallel_worker(worker_controls_string,DO_LOGGING,DO_DEBUGGING);
```
Comment (nickbattam-tessella) [inline admin/worker_4tests.m.template:32]: there is a lot of logic in this catch block and the ~ok block afterwards; can this be simplified or extracted to a helper function?

File: admin/worker_4tests.m.template [hunk 2]
```diff
@@ -13,14 +13,62 @@ function [ok,err_mess,je]=worker_4tests(worker_controls_string)
...
+catch ME1 % intercepted exception in processing failure or some odd bug indeed
+    pid = int64(feature('getpid'));
+    ok = false;
+    if exist('je','var') && isa(je,'JobExecutor') %exception thrown but JE is defined
+        if isnumeric(je.labIndex) && ~isempty(je.labIndex)
+            log_file = sprintf('WORKER_V2_pid%i_Node#%d_failure.log',pid,je.labIndex);
```
Comment (nickbattam-tessella) [inline admin/worker_4tests.m.template:38]: filename pattern should be a constant

File: admin/worker_4tests.m.template [hunk 3]
```diff
@@ -13,14 +13,62 @@ function [ok,err_mess,je]=worker_4tests(worker_controls_string)
...
+        else
+            log_file = sprintf('WORKER_V2_pid%i_JE_common_failure.log',pid);
```
Comment (nickbattam-tessella) [inline admin/worker_4tests.m.template:40]: filename pattern should be a constant

File: admin/worker_4tests.m.template [hunk 4]
```diff
@@ -13,14 +13,62 @@ function [ok,err_mess,je]=worker_4tests(worker_controls_string)
...
+        end
+    else
+        log_file = sprintf('WORKER_V2_pid%i_JE_FAIL_PROCESSING_failure.log',pid);
```
Comment (nickbattam-tessella) [inline admin/worker_4tests.m.template:43]: filename pattern should be a constant

File: admin/worker_4tests.m.template [hunk 5]
```diff
@@ -13,14 +13,62 @@ function [ok,err_mess,je]=worker_4tests(worker_controls_string)
...
-    % Set up non-default herbert or Horace path to test them
-    % herbert_on('/home/isis_direct_soft/Herbert_bugfix/herbert_core');
-    % horace_on('/home/isis_direct_soft/Horace_bugfix/horace_core');
+    try
+        horace_on()
+    catch ME
+        ok = false;
+        err_mess = ME;
+        pid = int64(feature('getpid'));
+        log_file = sprintf('WORKER_V2_pid%i_PackageInit_failure.log',pid);
```
Comment (nickbattam-tessella) [inline admin/worker_4tests.m.template:24]: filename pattern should be a constant

File: admin/worker_v2.m.template [hunk 6]
```diff
@@ -13,9 +13,62 @@ function [ok,err_mess,je]=worker_v2(worker_controls_string)
...
+je = [];
 if isempty(which('herbert_init.m'))
-    horace_on();
+    try
+        horace_on()
+    catch ME
+        ok = false;
+        err_mess = ME;
+        pid = int64(feature('getpid'));
+        log_file = sprintf('WORKER_V2_pid%i_PackageInit_failure.log',pid);
```
Comment (nickbattam-tessella) [inline admin/worker_v2.m.template:24]: filename pattern should be a constant (ideally shared with _4tests.template)

File: admin/worker_v2.m.template [hunk 7]
```diff
@@ -13,9 +13,62 @@ function [ok,err_mess,je]=worker_v2(worker_controls_string)
...
-[ok, err_mess,je] = parallel_worker(worker_controls_string,DO_LOGGING);
+DO_DEBUGGING = false;
+try
+    [ok, err_mess,je] = parallel_worker(worker_controls_string,DO_LOGGING,DO_DEBUGGING);
+catch ME1 % intercepted exception in processing failure or some odd bug indeed
+    pid = int64(feature('getpid'));
+    ok = false;
+    if exist('je','var') && isa(je,'JobExecutor') %exception thrown but JE is defined
+        if isnumeric(je.labIndex) && ~isempty(je.labIndex)
+            log_file = sprintf('WORKER_V2_pid%i_Node#%d_failure.log',pid,je.labIndex);
```
Comment (nickbattam-tessella) [inline admin/worker_v2.m.template:38]: filename pattern should be a constant (ideally shared with _4tests.template)

File: admin/worker_v2.m.template [hunk 8]
```diff
@@ -13,9 +13,62 @@ function [ok,err_mess,je]=worker_v2(worker_controls_string)
...
+        else
+            log_file = sprintf('WORKER_V2_pid%i_JE_common_failure.log',pid);
```
Comment (nickbattam-tessella) [inline admin/worker_v2.m.template:40]: filename pattern should be a constant (ideally shared with _4tests.template)

File: admin/worker_v2.m.template [hunk 9]
```diff
@@ -13,9 +13,62 @@ function [ok,err_mess,je]=worker_v2(worker_controls_string)
...
+        end
+    else
+        log_file = sprintf('WORKER_V2_pid%i_JE_FAIL_PROCESSING_failure.log',pid);
```
Comment (nickbattam-tessella) [inline admin/worker_v2.m.template:43]: filename pattern should be a constant (ideally shared with _4tests.template)

File: admin/worker_v2.m.template [hunk 10]
```diff
@@ -13,9 +13,62 @@ function [ok,err_mess,je]=worker_v2(worker_controls_string)
...
+    end
+    err_mess = ME1;
+end
+if ~ok && isa(err_mess,'MException')
+    pid = int64(feature('getpid'));
+    if isa(je,'JobExecutor')
+        log_file = sprintf('WORKER_V2_pid%i_Node#%d_failure.log',pid,je.labIndex);
```
Comment (nickbattam-tessella) [inline admin/worker_v2.m.template:50]: filename pattern should be a constant (ideally shared with _4tests.template)

File: admin/worker_v2.m.template [hunk 11]
```diff
@@ -13,9 +13,62 @@ function [ok,err_mess,je]=worker_v2(worker_controls_string)
...
+    elseif isempty(je)
+        log_file = sprintf('WORKER_V2_pid%i_INIT_failure.log',pid);
```
Comment (nickbattam-tessella) [inline admin/worker_v2.m.template:52]: filename pattern should be a constant (ideally shared with _4tests.template)

File: admin/worker_v2.m.template (hunk 7 above)
Comment (abuts) [reply admin/worker_v2.m.template]: as no logs are printed anyway, probably no point in complicating the logic

**Pull request merged**